library/src/com/google/maps/android/kml/KmlGroundOverlay.java

```diff
@@ -0,0 +1,112 @@
+package com.google.maps.android.kml;
+
+import com.google.android.gms.maps.model.GroundOverlayOptions;
+import com.google.android.gms.maps.model.LatLngBounds;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a KML Ground Overlay
+ */
+public class KmlGroundOverlay {
+
+    private final Map<String, String> mProperties;
+
+    private final GroundOverlayOptions mGroundOverlayOptions;
+
+    private String mImageUrl;
+
+    private LatLngBounds mLatLngBox;
+
+    /**
+     * Creates a new Ground Overlay
+     *
+     * @param imageUrl   url of the ground overlay image
+     * @param latLonBox  bounds of the image
+     * @param drawOrder  z index of the image
+     * @param visibility true if visible, false otherwise
+     * @param properties properties hashmap
+     * @param rotation   rotation of image
+     */
+    /* package */ KmlGroundOverlay(String imageUrl, LatLngBounds latLonBox, float drawOrder,
+            int visibility, HashMap<String, String> properties, float rotation) {
+        mGroundOverlayOptions = new GroundOverlayOptions();
+        mImageUrl = imageUrl;
+        mProperties = properties;
+        if (latLonBox == null) {
+            throw new IllegalArgumentException("No LatLonBox given");
+        }
+        mLatLngBox = latLonBox;
+        mGroundOverlayOptions.positionFromBounds(latLonBox);
+        mGroundOverlayOptions.bearing(rotation);
+        mGroundOverlayOptions.zIndex(drawOrder);
+        mGroundOverlayOptions.visible(visibility != 0);
+    }
+
+    /**
+     * Gets an image url
+     *
+     * @return An image url
+     */
+    public String getImageUrl() {
+        return mImageUrl;
+    }
+
+    /**
+     * Returns boundaries of the ground overlay
+     *
+     * @return Boundaries of the ground overlay
+     */
+    public LatLngBounds getLatLngBox() {
+        return mLatLngBox;
+    }
+
+    /**
+     * Gets an iterable of the properties
+     *
+     * @return Iterable of the properties
+     */
+    public Iterable getProperties() {
```

Comment:
Changed the implementation here - getProperties now returns the key set of all the properties (instead of the entry set)